Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,674 @@
+                        // If a plugin without a manifest failed to load due to an external issue (eg config),
+                        // this updates the manifest to the actual plugin values.
+                        manifest.Version = pluginStr;
+                        manifest.Name = plugin.Instance.Name;
+                        manifest.Description = plugin.Instance.Description;
+                        changed = true;
+                    }
+
+                    changed = changed || manifest.Status != PluginStatus.Active;
+                    manifest.Status = PluginStatus.Active;
+
+                    if (changed)
+                    {
+                        SaveManifest(manifest, plugin.Path);
+                    }
+                }
+
+                _logger.LogInformation("Loaded plugin: {PluginName} {PluginVersion}", plugin.Name, plugin.Version);
+
+                return instance;
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                _logger.LogError(ex, "Error creating {Type}", type.FullName);
+                if (plugin != null)
+                {
+                    if (ChangePluginState(plugin, PluginStatus.Malfunctioned))
+                    {
+                        _logger.LogInformation("Plugin {Path} has been disabled.", plugin.Path);
+                        return null;
+                    }
+                }
+
+                _logger.LogDebug("Unable to auto-disable.");
+                return null;
+            }
+        }
+
+        private void CheckIfStillSuperceded(LocalPlugin plugin)
+        {
+            if (plugin.Manifest.Status != PluginStatus.Superceded)
+            {
+                return;
+            }
+
+            var predecessor = _plugins.OrderByDescending(p => p.Version)
+                .FirstOrDefault(p => p.Id.Equals(plugin.Id) && p.IsEnabledAndSupported && p.Version != plugin.Version);
+            if (predecessor != null)
+            {
+                return;
+            }
+
+            plugin.Manifest.Status = PluginStatus.Active;
+        }
+
+        /// <summary>
+        /// Attempts to delete a plugin.
+        /// </summary>
+        /// <param name="plugin">A <see cref="LocalPlugin"/> instance to delete.</param>
+        /// <returns>True if successful.</returns>
+        private bool DeletePlugin(LocalPlugin plugin)
+        {
+            // Attempt a cleanup of old folders.
+            try
+            {
+                Directory.Delete(plugin.Path, true);
+                _logger.LogDebug("Deleted {Path}", plugin.Path);
+                _plugins.Remove(plugin);
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                return false;
```

Comment:
Should log the exception here

Reply:
I purposefully left this error message out, as the system is expecting this error (especially on windows), and a tailored error message is reported from the calling method.

Didn't want the user reporting errors from the logs when it's the norm state of play.